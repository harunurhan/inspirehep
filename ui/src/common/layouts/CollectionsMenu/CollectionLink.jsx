import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { Tag } from 'antd';

import './CollectionLink.scss';
import styleVariables from '../../../styleVariables';

const GREEN = styleVariables['success-color'];

function CollectionLink({ to, active, children, newCollection }) {
  return (
    <span className="__CollectionLink__ mh4 m-mh2">
      <Link
        className={classNames('link f5 white', {
          active,
        })}
        to={to}
      >
        {children}
      </Link>
      {newCollection && (
        <Tag className="new-tag" color={GREEN}>
          New
        </Tag>
      )}
    </span>
  );
}

CollectionLink.propTypes = {
  to: PropTypes.string.isRequired,
  active: PropTypes.bool,
  newCollection: PropTypes.bool,
  children: PropTypes.node,
};

export default CollectionLink;
