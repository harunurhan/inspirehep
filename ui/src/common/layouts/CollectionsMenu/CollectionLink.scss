.__CollectionLink__ {
  padding-bottom: 0.2em;

  .link:hover,
  .link.active {
    $color: white;
    color: $color;
    border-bottom: solid 0.15em $color;
  }

  .new-tag {
    margin-left: 8px;
    padding: 0 2px;
    line-height: 0.9rem;
    font-size: 0.6rem;
    font-weight: 600;
    cursor: unset;
    &:hover {
      opacity: 1;
    }
  }
}
