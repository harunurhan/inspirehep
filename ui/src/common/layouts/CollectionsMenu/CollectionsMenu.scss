.__CollectionsMenu__ {
  width: 100%;
  padding-bottom: 0.3em;

  .collection-link {
    padding-bottom: 0.2em;

    &:hover,
    &:active {
      $color: white;
      color: $color;

      @media (min-width: $screen-xs-max) {
        border-bottom: solid 0.15em $color;
      }
    }
  }
}
