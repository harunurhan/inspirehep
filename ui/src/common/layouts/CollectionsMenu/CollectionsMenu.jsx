import React from 'react';
import { Row, Col } from 'antd';
import { Link } from 'react-router-dom'

import './CollectionsMenu.scss';
import { LITERATURE, AUTHORS, JOBS, CONFERENCES } from '../../routes';

// eslint-disable-next-line react/prop-types
function CollectionLink({ to, children }) {
  return (
    <Link className="collection-link mh4 f4 white" to={to}>
      {children}
    </Link>
  );
}


function CollectionsMenu() {
  return (
    <Row className="__CollectionsMenu__" justify="center">
      <Col>
        <CollectionLink to={LITERATURE}>Literature</CollectionLink>
      </Col>
      <Col>
        <CollectionLink to={AUTHORS}>Authors</CollectionLink>
      </Col>
      <Col>
        <CollectionLink to={JOBS}>Jobs</CollectionLink>
      </Col>
      <Col>
        <CollectionLink to={CONFERENCES}>Conferences</CollectionLink>
      </Col>
    </Row>
  )
}

export default CollectionsMenu;
