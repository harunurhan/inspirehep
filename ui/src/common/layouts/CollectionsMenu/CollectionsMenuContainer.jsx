import { connect } from 'react-redux';

import CollectionsMenu from './CollectionsMenu';

const stateToProps = state => ({
  currentPathname: state.router.location.pathname,
});

export default connect(stateToProps)(CollectionsMenu);
