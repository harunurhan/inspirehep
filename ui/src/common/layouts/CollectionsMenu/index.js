import CollectionsMenu from './CollectionsMenu';

export default CollectionsMenu;