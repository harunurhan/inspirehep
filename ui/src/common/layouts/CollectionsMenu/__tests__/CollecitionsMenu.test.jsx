import React from 'react';
import { shallow } from 'enzyme';

import CollectionsMenu from '../CollectionsMenu';

describe('CollectionsMenu', () => {
  it('renders', () => {
    const wrapper = shallow(
      <CollectionsMenu />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
