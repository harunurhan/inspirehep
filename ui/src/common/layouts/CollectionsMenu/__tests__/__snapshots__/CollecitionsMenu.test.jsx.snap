// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsMenu renders when authors page 1`] = `
<Row
  activeCollection="authors"
  className="__CollectionsMenu__"
  gutter={0}
  justify="center"
>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/literature"
    >
      Literature
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white active"
      to="/authors"
    >
      Authors
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/jobs"
    >
      Jobs
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/conferences"
    >
      Conferences
    </Link>
  </Col>
</Row>
`;

exports[`CollectionsMenu renders when conferences page 1`] = `
<Row
  activeCollection="conferences"
  className="__CollectionsMenu__"
  gutter={0}
  justify="center"
>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/literature"
    >
      Literature
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/authors"
    >
      Authors
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/jobs"
    >
      Jobs
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white active"
      to="/conferences"
    >
      Conferences
    </Link>
  </Col>
</Row>
`;

exports[`CollectionsMenu renders when home page 1`] = `
<Row
  activeCollection=""
  className="__CollectionsMenu__"
  gutter={0}
  justify="center"
>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/literature"
    >
      Literature
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/authors"
    >
      Authors
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/jobs"
    >
      Jobs
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/conferences"
    >
      Conferences
    </Link>
  </Col>
</Row>
`;

exports[`CollectionsMenu renders when jobs page 1`] = `
<Row
  activeCollection="jobs"
  className="__CollectionsMenu__"
  gutter={0}
  justify="center"
>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/literature"
    >
      Literature
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/authors"
    >
      Authors
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white active"
      to="/jobs"
    >
      Jobs
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/conferences"
    >
      Conferences
    </Link>
  </Col>
</Row>
`;

exports[`CollectionsMenu renders when literature page 1`] = `
<Row
  activeCollection="literature"
  className="__CollectionsMenu__"
  gutter={0}
  justify="center"
>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white active"
      to="/literature"
    >
      Literature
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/authors"
    >
      Authors
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/jobs"
    >
      Jobs
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/conferences"
    >
      Conferences
    </Link>
  </Col>
</Row>
`;

exports[`CollectionsMenu renders when submissions page 1`] = `
<Row
  activeCollection="submissions"
  className="__CollectionsMenu__"
  gutter={0}
  justify="center"
>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/literature"
    >
      Literature
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/authors"
    >
      Authors
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/jobs"
    >
      Jobs
    </Link>
  </Col>
  <Col>
    <Link
      className="collection-link mh4 sm-mh2 f5 white"
      to="/conferences"
    >
      Conferences
    </Link>
  </Col>
</Row>
`;
